src/mahalanobis.jl

```diff
@@ -1,19 +1,19 @@
 # Mahalanobis distances
 
-struct Mahalanobis{T} <: Metric
-    qmat::AbstractMatrix{T}
+struct Mahalanobis{T,M<:AbstractMatrix{T}} <: Metric
+    qmat::M
 end
 
-struct SqMahalanobis{T} <: SemiMetric
-    qmat::AbstractMatrix{T}
+struct SqMahalanobis{T,M<:AbstractMatrix{T}} <: SemiMetric
+    qmat::M
 end
```

Comment:
These changes are the only necessary.

Reply:
I think we can also eliminate the mentions to `T` in function definitions @dkarrasch given that they are not used in the function body. Also, eliminate the premature `T<:Real` that appears in some of them. Ideally we leave it free to be any type as people may want to use the distance with exotic entry types.